contracts/light-clients/AttestorLightClient.sol

```diff
@@ -0,0 +1,216 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.28;
+
+import { AccessControl } from "@openzeppelin-contracts/access/AccessControl.sol";
+import { ECDSA } from "@openzeppelin-contracts/utils/cryptography/ECDSA.sol";
+
+import { ILightClient } from "../interfaces/ILightClient.sol";
+import { ILightClientMsgs } from "../msgs/ILightClientMsgs.sol";
+import { IAttestorLightClient } from "./IAttestorLightClient.sol";
+import { IAttestorLightClientErrors } from "./errors/IAttestorLightClientErrors.sol";
+import { IAttestorMsgs } from "./msgs/IAttestorMsgs.sol";
+import { IAttestorLightClientMsgs } from "./msgs/IAttestorLightClientMsgs.sol";
+
+/// @title Attestor-based IBC Light Client
+/// @notice Implements an IBC light client that trusts an off-chain m-of-n attestor set.
+contract AttestorLightClient is IAttestorLightClient, IAttestorLightClientErrors, AccessControl {
+    /// @notice Current attestor-set configuration and latest trusted height/frozen flag.
+    IAttestorLightClientMsgs.ClientState private clientState;
+    /// @notice Tracks whether an `attestor` address is part of the configured attestor set.
+    mapping(address attestor => bool isAttestor) private _isAttestor;
```

Comment:
So I guess the idea is that we will simply create a new light client if we want to change the attestor set. I am fine with this

Reply:
Yeah, at least for now, I think it's the easiest path